### PR TITLE
virsh_maxvcpus: Fix maxvcpu comparision method

### PR DESCRIPTION
**Pull request by dzhengfy**
According to BZ 1455363 comment #10, we should compare virsh.maxvcpus to
virsh.domcapabilities instead of virsh.capabilities for max vcpu number.
The old way is inaccurate for PPC case. So this is to fix the problem.

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/host/virsh_maxvcpus.py [hunk 1]
```diff
@@ -69,21 +68,21 @@ def run(test, params, env):
     if libvirt_version.version_compare(2, 3, 0):
         try:
             maxvcpus = None
+            dom_capabilities = None
             # make sure we take maxvcpus from right host, helps incase remote
-            virsh_dargs = {'uri': connect_uri}
-            virsh_instance = virsh.Virsh(virsh_dargs)
             try:
-                capa = capability_xml.CapabilityXML(virsh_instance)
```
Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/host/virsh_maxvcpus.py]: so the 'virsh_instance' is useless?

File: libvirt/tests/src/virsh_cmd/host/virsh_maxvcpus.py [hunk 2]
```diff
@@ -69,21 +68,21 @@ def run(test, params, env):
...
-                host_arch = capa.arch
-                maxvcpus = capa.get_guest_capabilities()['hvm'][host_arch]['maxcpus']
-                logging.debug("maxvcpus calculate is %s", maxvcpus)
+                dom_capabilities = domcap.DomCapabilityXML()
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/host/virsh_maxvcpus.py]: I think we need virsh_instance,
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/avocado_plugins_vt-51.0-py2.7.egg/virttest/propcan.py", line 230, in __getattr__
    return super(PropCanBase, self).__getattribute__(key)
AttributeError: 'DomCapabilityXML' object has no attribute 'max'

```

File: libvirt/tests/src/virsh_cmd/host/virsh_maxvcpus.py (hunk 1 above)
Comment (sathnaga) [inline libvirt/tests/src/virsh_cmd/host/virsh_maxvcpus.py]: virsh_instance is needed incase of remote uri testcases

**Comment by dzhengfy** [comment]
@sathnaga , I did test remote cases. The result shows 'virsh_instance' is useless for remote uri cases.
  # avocado run --vt-type libvirt virsh.maxvcpus.connect_to_remote
JOB ID     : 9c7343b5531028b1f64297f10d8db289147c8b70
JOB LOG    : /root/avocado/job-results/job-2017-07-19T10.49-9c7343b/job.log
TESTS      : 9
 (1/9) type_specific.io-github-autotest-libvirt.virsh.maxvcpus.connect_to_remote.no_option: PASS (5.01 s)
 (2/9) type_specific.io-github-autotest-libvirt.virsh.maxvcpus.connect_to_remote.with_option.with_qemu: PASS (11.97 s)
 (3/9) type_specific.io-github-autotest-libvirt.virsh.maxvcpus.connect_to_remote.with_option.with_qemu1: PASS (11.46 s)
 (4/9) type_specific.io-github-autotest-libvirt.virsh.maxvcpus.connect_to_remote.with_option.with_kvm: PASS (11.20 s)
 (5/9) type_specific.io-github-autotest-libvirt.virsh.maxvcpus.connect_to_remote.with_option.with_kvm1: PASS (10.76 s)
 (6/9) type_specific.io-github-autotest-libvirt.virsh.maxvcpus.connect_to_remote.with_option.with_kqemu: PASS (12.41 s)
 (7/9) type_specific.io-github-autotest-libvirt.virsh.maxvcpus.connect_to_remote.with_option.with_kqemu1: PASS (12.30 s)
 (8/9) type_specific.io-github-autotest-libvirt.virsh.maxvcpus.connect_to_remote.unexpect_option.with_xyz: PASS (11.67 s)
 (9/9) type_specific.io-github-autotest-libvirt.virsh.maxvcpus.connect_to_remote.unexpect_option.with_xyz1: PASS (13.68 s)
RESULTS    : PASS 9 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0
JOB HTML   : /root/avocado/job-results/job-2017-07-19T10.49-9c7343b/html/results.html
TESTS TIME : 100.46 s

**Comment by sathnaga** [comment]
@dzhengfy Cool then 'Ack' from me.

File: libvirt/tests/src/virsh_cmd/host/virsh_maxvcpus.py [hunk 3]
```diff
@@ -69,21 +68,21 @@ def run(test, params, env):
...
             except:
                 raise exceptions.TestFail("Failed to get maxvcpus from "
-                                          "capabilities xml\n%s" % capa)
+                                          "domcapabilities xml:\n%s"
+                                          % dom_capabilities)
             if not maxvcpus:
-                raise exceptions.TestFail("Failed to get guest section for "
-                                          "host arch: %s from capabilities "
-                                          "xml\n%s" % (host_arch, capa))
+                raise exceptions.TestFail("Failed to get max value for vcpu"
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/host/virsh_maxvcpus.py]: please use test.fail() as suggested in https://github.com/avocado-framework/avocado-vt/pull/966/files, can you help in changing in other places too ?

Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/host/virsh_maxvcpus.py]: @balamuruhans sure, will do.

**Comment by sathnaga** [comment]
@balamuruhans can we merge this  as it fixes a issue and take the rest of changes as separate PR.

**Comment by sathnaga** [comment]
@balamuruhans Thanks :-), lets merge this.

**Pull request merged**